pkg/kubelet/volumemanager/populator/desired_state_of_world_populator.go

```diff
@@ -396,17 +397,16 @@ func mountedReadOnlyByPod(podVolume v1.Volume, pod *v1.Pod) bool {
 	if podVolume.PersistentVolumeClaim.ReadOnly {
 		return true
 	}
-	for _, container := range pod.Spec.InitContainers {
-		if !mountedReadOnlyByContainer(podVolume.Name, &container) {
-			return false
-		}
-	}
-	for _, container := range pod.Spec.Containers {
-		if !mountedReadOnlyByContainer(podVolume.Name, &container) {
+
+	mountedReadOnly := true
+	podutil.VisitContainers(&pod.Spec, func(c *v1.Container) bool {
```

Comment:
cc volumemanager approvers @msau42 @saad-ali @jsafrane

Reply:
Can this be simplified to just `return podutil.VisitContainers(...)`?

Reply:
I was worried that conflating the two levels of bool returns for "visiting was completed" and "mounted read only by pod" would be confusing, but I tried it out and it makes sense for this "search for a counterexample" sort of pattern.

I've updated to this, thanks for the suggestion.